packages/flutter/test/widgets/selectable_region_test.dart

```diff
@@ -310,6 +310,86 @@ void main() {
       expect(edgeEvent.globalPosition, const Offset(200.0, 50.0));
     });
 
+  testWidgets(
+    'touch long press cancel does not send ClearSelectionEvent',
+    (WidgetTester tester) async {
+      final UniqueKey spy = UniqueKey();
+      await tester.pumpWidget(MaterialApp(
+        home: SelectableRegion(
+          focusNode: FocusNode(),
+          selectionControls: materialTextSelectionControls,
+          child: SelectionSpy(key: spy),
+        ),
+      ));
+      await tester.pumpAndSettle();
+
+      final RenderSelectionSpy renderSelectionSpy =
+          tester.renderObject<RenderSelectionSpy>(find.byKey(spy));
+      renderSelectionSpy.events.clear();
+      final TestGesture gesture =
+          await tester.startGesture(const Offset(200.0, 200.0));
+
+      addTearDown(gesture.removePointer);
+
+      await tester.pump(const Duration(milliseconds: 500));
+      await gesture.cancel();
+      expect(
+        renderSelectionSpy.events.any((SelectionEvent element) => element is ClearSelectionEvent),
+        isFalse,
+      );
+    },
+  );
+
+    testWidgets(
+      'scrolling after the selection does not send ClearSelectionEvent',
+      (WidgetTester tester) async {
+        // Regression test for https://github.com/flutter/flutter/issues/128765
+        final UniqueKey spy = UniqueKey();
+        await tester.pumpWidget(
+          MaterialApp(
+            home: SizedBox(
+              height: 750,
+              child: SingleChildScrollView(
+                child: SizedBox(
+                  height: 2000,
+                  child: SelectableRegion(
+                    focusNode: FocusNode(),
+                    selectionControls: materialTextSelectionControls,
+                    child: SelectionSpy(key: spy),
+                  ),
+                ),
+              ),
+            ),
+          ),
+        );
+        await tester.pumpAndSettle();
+
+        final RenderSelectionSpy renderSelectionSpy =
+            tester.renderObject<RenderSelectionSpy>(find.byKey(spy));
+        final TestGesture selectGesture =
+            await tester.startGesture(const Offset(200.0, 200.0));
+        addTearDown(selectGesture.removePointer);
+        await tester.pump(const Duration(milliseconds: 500));
+
```

Comment:
Maybe here instead of moving immediately after the long press we should just do `selectGesture.up()` and check for the `SelectWordSelectionEvent`. Then clear the events list with `renderSelectionSpy.events.clear()`. Then try a scroll gesture to see that `ClearSelectionEvent` is not sent. This would simulate taking our finger off the screen and then initiating a scroll.

Reply:
Thank you @Renzo-Olivares I have updated the PR accordingly.